### PR TITLE
Fix TypeScript Definitions for making lit-element happy

### PR DESCRIPTION
**Pull request by mistyharsh**
This PR fixes issue #37.

### CODE REVIEW

File: packages/mixins/apollo-element-mixin.d.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import ApolloClient, { ApolloError } from 'apollo-client';
 import { DocumentNode } from 'graphql';
+import { LitElement } from 'lit-element';
```
Comment (bennypowers) [inline packages/mixins/apollo-element-mixin.d.ts]: The mixin packages shuold not reference lit-element, as they are used for a variety of packages. we'll need to find another solution.

File: packages/lit-apollo/apollo-query.d.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { ApolloQuery as ApolloQueryBase } from "@apollo-elements/mixins/apollo-mutation-mixin";
+import { ApolloMutation as ApolloQueryBase } from "@apollo-elements/mixins/apollo-mutation-mixin";
```
Comment (bennypowers) [inline packages/lit-apollo/apollo-query.d.ts]: Why should this be ApolloMutation? it's the query mixin.. probably should be
```suggestion
import { ApolloQuery as ApolloQueryBase } from "@apollo-elements/mixins/apollo-query-mixin";
```

**Comment by bennypowers** [comment]
Thank you so much for the PR. Please see review comments.

**Comment by mistyharsh** [comment]
@bennypowers Thank you for the review. I did think about these issues before raising the PR. I just hoped to fix TypeScript issues quickly. I will work it out these comments and see how we can adhere to the design.

**Comment by mistyharsh** [comment]
So there are a couple of things:

  - After reading [Ander's comment](https://github.com/Microsoft/TypeScript/issues/4890#issuecomment-141879451), It seems it is almost impossible to solve this problem. The only option is to use class and interface merging.
  - `ApolloMutation` cannot be currently fixed unless we change `update` property as it conflicts with `LitElement.update()` method. We should implement RFC #35.
  - With declaration merging, we get empty interfaces which fail the linting rules.

**Comment by bennypowers** [comment]
I'm ok with implementing #35, although it will represent a major release.

What I'd *love* to see would be some tests for these typings.

Another option we should consider is using [JSDoc inline typings instead of .d.ts files](https://dev.to/dakmor/type-safe-web-components-with-jsdoc-4icf).

**Comment by mistyharsh** [comment]
Having Tests for typescript definition files is not very common. We can treat our files and run `tsc` to validate our type definitions the way we run linter for our JS code. That is something I can think of.

As far as JSDoc is concerned, it looks interesting but usually ends up being cumbersome to write than plain TS definitions.

**Comment by RSWilli** [comment]
what is the status for this? I am having the same issue with TS and lit-apollo.

This PR doesn't just "make lit-element happy", currently TS just doesn't compile the code because it complains that none of the apollo properties (client, query, etc.) are present on my class

**Comment by bennypowers** [comment]
@RSWilli can you send me a repro to tinker with? My volunteer time is short this week as I just moved apartment, but I'm really hoping to solve this for you ASAP, so if you have a ready made example repo, or some changes for this PR I'll try my best to get something out.

**Comment by RSWilli** [comment]
Repro: https://stackblitz.com/edit/typescript-hfggxi

for me, editing https://github.com/apollo-elements/apollo-elements/blob/master/packages/lit-apollo/apollo-query.d.ts#L1 from @apollo-elements/mixins/apollo-mutation-mixin to @apollo-elements/mixins/apollo-query-mixin fixes it, but I don't know if it brakes anything else that my usecase does not include

**Comment by bennypowers** [comment]
very nice work, thanks

**Pull request merged**